### PR TITLE
fix(pointcloud_preprocessor): fix arrayIndexThenCheck warning

### PR DESCRIPTION
**Pull request by veqcc**
## Description

This is a fix based on cppcheck `arrayIndexThenCheck` warning

```
sensing/pointcloud_preprocessor/src/outlier_filter/dual_return_outlier_filter_nodelet.cpp:234:36: style: Array index 'current_deleted_index' is used before limits check. [arrayIndexThenCheck]
      while ((uint)deleted_azimuths[current_deleted_index] <
                                   ^
```

You know the current implementation is correct.
This fix is just to avoid cppcheck warning by changing the evaluation order in while loop condition.

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: sensing/pointcloud_preprocessor/src/outlier_filter/dual_return_outlier_filter_nodelet.cpp [hunk 1]
```diff
@@ -231,10 +231,10 @@ void DualReturnOutlierFilterComponent::filter(
       if (deleted_azimuths.size() == 0) {
         continue;
       }
-      while ((uint)deleted_azimuths[current_deleted_index] <
+      while (current_deleted_index < (deleted_azimuths.size() - 1) &&
```
Comment (vividf) [inline sensing/pointcloud_preprocessor/src/outlier_filter/dual_return_outlier_filter_nodelet.cpp:234]: Isn't it `current_deleted_index < deleted_azimuths.size() `

**Review by vividf — CHANGES_REQUESTED** [review]

**Comment by vividf** [comment]
@badai-nguyen 
The original code probably has a the bug of` current_deleted_index < (deleted_azimuths.size() - 1)`, which is weird if the size is equal to 1. Is this a bug?

**Review by badai-nguyen — APPROVED** [review]
@vividf @veqcc 
I guess the reason is that in the original order, `current_deleted_index < (deleted_azimuths.size() - 1)` seems more correct to avoid `out of range` error in the next loop since after that `current_deleted_index` is incremented. 
After change the conditional order, `current_deleted_index < deleted_azimuths.size() ` should be fine.

File: sensing/pointcloud_preprocessor/src/outlier_filter/dual_return_outlier_filter_nodelet.cpp (hunk 1 above)
Comment (veqcc) [reply sensing/pointcloud_preprocessor/src/outlier_filter/dual_return_outlier_filter_nodelet.cpp]: > Please let the person who made the comment resolve the conversation (in addition to being good manners help keep track of what has been handled / answered in the PR)

OK, thank you for your advice :+1:

**Review by vividf — APPROVED** [review]
LGTM!

**Pull request merged**